src/ProtocolFees.sol

```diff
@@ -49,19 +49,27 @@ abstract contract ProtocolFees is IProtocolFees, PausableRole {
             // in mind.
             if (gasleft() < controllerGasLimit) revert ProtocolFeeCannotBeFetched();
 
-            (bool _success, bytes memory _data) = address(protocolFeeController).call{gas: controllerGasLimit}(
-                abi.encodeCall(IProtocolFeeController.protocolFeeForPool, (key))
-            );
-            // Ensure that the return data fits within a word
-            if (!_success || _data.length > 32) return (false, 0);
-
+            uint256 gasLimit = controllerGasLimit;
+            address targetProtocolFeeController = address(protocolFeeController);
+            bytes memory data = abi.encodeCall(IProtocolFeeController.protocolFeeForPool, (key));
             uint256 returnData;
             assembly ("memory-safe") {
-                returnData := mload(add(_data, 0x20))
+                success := call(gasLimit, targetProtocolFeeController, 0, add(data, 0x20), mload(data), 0, 0)
+
+                // success if return data size is 32 bytes
+                // only load the return value if it is 32 bytes to prevent gas griefing
+                success := and(success, eq(returndatasize(), 32))
+
+                // load the return data if success is true
+                if success {
+                    let fmp := mload(0x40)
```

Comment:
Should we use this instead?

https://github.com/nomad-xyz/ExcessivelySafeCall/blob/main/src/ExcessivelySafeCall.sol#L55-L57

Reply:
Ok I see that this is the memory-safe implementation (using fmp), which the nomad-xyz contract does not implement. Functionally they should be the same.